portal-web/test/functional/com/liferay/portalweb/macros/headless/headlessadminuser/MakeARelationshipWithSystemObjectThroughAPI.testcase

```diff
@@ -0,0 +1,443 @@
+
+			UserAccountAPI.assertResponseHasCorrectRelatedEntryName(
+				customObjectEntryId = "${foundationId2}",
+				expectedValue = "Foundation Second",
+				relationshipName = "usersFoundations",
+				userAccountId = "${userAccountId2}");
+		}
+	}
+
+	@disable-webdriver = "true"
+	@priority = "4"
+	test CanDeleteCustomObjectEntryRelatedToSystemObjectByOneToMany {
+		property portal.acceptance = "true";
+
+		task ("Given a foundation object") {
+			var objectDefinitionId2 = ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_label = "foundation",
+				en_US_plural_label = "foundations",
+				name = "Foundation",
+				requiredStringFieldName = "name");
+		}
+
+		task ("Given oneToMany relationship usersFoundations created") {
+			var objectDefinitionId1 = ObjectDefinitionAPI.getObjectDefinitionIdByName(name = "User");
+
+			var relationshipName = ObjectDefinitionAPI.createRelationship(
+				deletionType = "cascade",
+				en_US_label = "UsersFoundations",
+				name = "usersFoundations",
+				objectDefinitionId1 = "${objectDefinitionId1}",
+				objectDefinitionId2 = "${objectDefinitionId2}",
+				type = "oneToMany");
+		}
+
+		task ("Given two foundation entries created") {
+			var foundationId1 = ObjectDefinitionAPI.createObjectEntryWithName(
+				en_US_plural_label = "foundations",
+				name = "Foundation First");
+			var foundationId2 = ObjectDefinitionAPI.createObjectEntryWithName(
+				en_US_plural_label = "foundations",
+				name = "Foundation Second");
+		}
+
+		task ("Given the userAccount entry created") {
+			UserAccountAPI.setUpGlobalUserAccountId(
+				alternateName = "user1",
+				emailAddress = "user1@liferay.com",
+				familyName = "userfn1",
+				givenName = "usergn1");
+
+			var userAccountId1 = "${staticUserAccountId}";
+		}
+
+		task ("Given I relate all the foundation entries with the userAccount entry through the userAccount PUT endpoint") {
+			UserAccountAPI.relateObjectEntries(
+				customObjectId = "${foundationId1}",
+				relationshipName = "usersFoundations",
+				userAccountId = "${userAccountId1}");
+
+			UserAccountAPI.relateObjectEntries(
+				customObjectId = "${foundationId2}",
+				relationshipName = "usersFoundations",
+				userAccountId = "${userAccountId1}");
+		}
+
+		task ("When deleting one of the foundation entries") {
+			ObjectDefinitionAPI.deleteObjectEntry(
+				en_US_plural_label = "foundations",
+				objectEntryId = "${foundationId1}");
+		}
+
+		task ("Then I can see only one entry correctly related in GET request response") {
+			UserAccountAPI.assertResponseHasCorrectRelatedEntryName(
+				customObjectEntryId = "${foundationId2}",
+				expectedValue = "Foundation Second",
+				relationshipName = "usersFoundations",
+				userAccountId = "${userAccountId1}");
+		}
+	}
+
+	@disable-webdriver = "true"
+	@priority = "4"
+	test CanRelateCustomObjectEntriesToSystemObjectEntries {
+		property portal.acceptance = "true";
+
+		task ("Given a firstFoundation object") {
```

Comment:
why don't we use the foundation object here as well? this way we could move all the foundation creation to the setup and we would start this test with creating secondFoundation